src/compiler/checker.ts

```diff
@@ -383,20 +383,42 @@ namespace ts {
             // return undefined if we can't find a symbol.
         }
 
-        /** Returns true if node1 is defined before node 2**/
-        function isDefinedBefore(node1: Node, node2: Node): boolean {
-            let file1 = getSourceFileOfNode(node1);
-            let file2 = getSourceFileOfNode(node2);
+        const enum RelativeLocation {
+            Unknown,
+            SameFileLocatedBefore,
+            SameFileLocatedAfter,
+            DifferentFilesLocatedBefore,
+            DifferentFilesLocatedAfter,

```

Comment:
Is this Before and After based on file order in the compilation? Are these only determined when -out is specified?